app/src/ui/stashing/confirm-discard-stash.tsx

```diff
@@ -0,0 +1,78 @@
+import React = require('react')
+import { Dialog, DialogContent, DialogFooter } from '../dialog'
+import { Repository } from '../../models/repository'
+import { Dispatcher } from '../dispatcher'
+import { ButtonGroup } from '../lib/button-group'
+import { Button } from '../lib/button'
+import { Row } from '../lib/row'
+import { IStashEntry } from '../../models/stash-entry'
+
+interface IDiscardStashProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly stash: IStashEntry
+  readonly onDismissed: () => void
+}
+
+interface IDiscardStashState {
+  readonly isDiscarding: boolean
+}
+/**
+ * Dialog to confirm dropping a stash
+ */
+export class DiscardStash extends React.Component<
+  IDiscardStashProps,
+  IDiscardStashState
+> {
+  public constructor(props: IDiscardStashProps) {
+    super(props)
+
+    this.state = {
+      isDiscarding: false,
+    }
+  }
+
+  public render() {
+    const title = __DARWIN__ ? 'Discard Stash?' : 'Discard stash?'
+
+    return (
+      <Dialog
+        id="discard-stash"
+        type="warning"
+        title={title}
+        loading={this.state.isDiscarding}
+        disabled={this.state.isDiscarding}
+        onSubmit={this.onSubmit}
+        onDismissed={this.props.onDismissed}
+      >
+        <DialogContent>
+          <Row>Are you sure you want to discard these stashed changes?</Row>
+        </DialogContent>
+        <DialogFooter>
+          <ButtonGroup>
+            <Button type="submit">Discard</Button>
```

Comment:
I think the type="submit" should be on the `Cancel` button to make that the blue default button on macOS, shouldn't it?
<img width="436" alt="Screen Shot 2019-04-23 at 8 15 04 AM" src="https://user-images.githubusercontent.com/5091167/56588401-7f836100-65a0-11e9-9319-95cb71fc24b9.png">

Reply:
9fd383e